### PR TITLE
Expanded access control section from cairo

### PR DESCRIPTION
**Pull request by ggrieco-tob**

### CODE REVIEW

**Comment by montyly** [comment]
@ggrieco-tob : can you take a look at the comments above?

@smonicas : can you take a look at the PR?

File: not-so-smart-contracts/cairo/access_controls/README.md [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
 
-It is still possible to interact with contracts directly. But from the perspective of the contract, the caller's address will be 0. Since 0 is also the default value for uninitialized storage, it's possible to accidentally construct access control checks that fail open instead of properly restricting access to only the intended users.
+* If a contract is called from another L2 contract, [access control should be implemented using `get_caller_address()`](https://www.cairo-lang.org/docs/hello_starknet/user_auth.html#getting-the-caller-address).
+
+* If a contract has a function that should be called from L1 contract, [the `@l1_handler` should be used](https://starknet.io/docs/hello_starknet/l1l2.html#receiving-a-message-from-l1). This forces the first parameter of such function to be the address of the contract or EOA that performed the cross-chain call.
+
+Corner cases to consider:
```
[the remark this answers is not in the document]
Comment (ggrieco-tob) [reply not-so-smart-contracts/cairo/access_controls/README.md]: Fixed

File: not-so-smart-contracts/cairo/access_controls/README.md [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
 
-It is still possible to interact with contracts directly. But from the perspective of the contract, the caller's address will be 0. Since 0 is also the default value for uninitialized storage, it's possible to accidentally construct access control checks that fail open instead of properly restricting access to only the intended users.
+* If a contract is called from another L2 contract, [access control should be implemented using `get_caller_address()`](https://www.cairo-lang.org/docs/hello_starknet/user_auth.html#getting-the-caller-address).
+
+* If a contract has a function that should be called from L1 contract, [the `@l1_handler` should be used](https://starknet.io/docs/hello_starknet/l1l2.html#receiving-a-message-from-l1). This forces the first parameter of such function to be the address of the contract or EOA that performed the cross-chain call.
```
[the remark this answers is not in the document]
Comment (ggrieco-tob) [reply not-so-smart-contracts/cairo/access_controls/README.md]: Moved!

**Comment by ggrieco-tob** [comment]
@montyly this is ready for another review

File: not-so-smart-contracts/cairo/access_controls/README.md [hunk 3]
```diff
@@ -1,8 +1,12 @@
...
 
-It is still possible to interact with contracts directly. But from the perspective of the contract, the caller's address will be 0. Since 0 is also the default value for uninitialized storage, it's possible to accidentally construct access control checks that fail open instead of properly restricting access to only the intended users.
+* If a contract is called from another L2 contract, [access control should be implemented using `get_caller_address()`](https://www.cairo-lang.org/docs/hello_starknet/user_auth.html#getting-the-caller-address).
```
Comment (montyly) [inline not-so-smart-contracts/cairo/access_controls/README.md:5]: can we explain why in one sentence?

Comment (ggrieco-tob) [reply not-so-smart-contracts/cairo/access_controls/README.md]: Done, please take a look

**Comment by montyly** [comment]
@smonicas or @technovision99 : can you take a look at the PR?

**Comment by technovision99** [comment]
lgtm, after regenesis we should link here instead imo: https://docs.starknet.io/documentation/architecture_and_concepts/L1-L2_Communication/messaging-mechanism/

**Comment by montyly** [comment]
replaced by https://github.com/crytic/building-secure-contracts/pull/338

**Pull request closed without merging**